python/ray/data/datasource/file_based_datasource.py

```diff
@@ -248,7 +248,7 @@ def read_files(
                     partitions = parse(read_path)
 
                 with RetryingContextManager(
```

Comment:
I think this class name is a bit too general, what about updating it to `RetryingPyarrowFileContext`